### PR TITLE
Ddcyls 1628

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

**Comment by chandraAndem** [comment]
Test this please

File: it/uk/gov/hmrc/traderservices/services/MongoDBCachedJourneyServiceISpec.scala [hunk 1]
```diff
@@ -1,28 +1,20 @@
...
-import org.mongodb.scala.MongoDatabase
-import uk.gov.hmrc.mongo.CurrentTimestampSupport
-import uk.gov.hmrc.mongo.cache.CacheItem
-import uk.gov.hmrc.mongo.cache.DataKey
-import java.time.Instant
-import play.api.libs.json.Json
-import play.api.libs.json.JsValue
-import play.api.libs.json.JsString
 
 class MongoDBCachedJourneyServiceISpec extends MongoDBCachedJourneyServiceISpecSetup {
```
Comment (ashwinirudrappa) [inline it/uk/gov/hmrc/traderservices/services/MongoDBCachedJourneyServiceISpec.scala:19]: As the file MongoDBCachedJourneyService.scala file was deleted, please consider using the correct name of the file and replace of the word MongoDBCachedJourneyService with the correct one.

**Review by ashwinirudrappa — COMMENTED** [review]
Please look into it the comments

File: app/uk/gov/hmrc/traderservices/controllers/AmendCaseJourneyController.scala [hunk 2]
```diff
@@ -17,25 +17,30 @@
...
 import uk.gov.hmrc.traderservices.wiring.AppConfig
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
 
 @Singleton
 class AmendCaseJourneyController @Inject() (
+  sessionStateService: SessionStateService,
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/uk/gov/hmrc/traderservices/controllers/AmendCaseJourneyController.scala]: Done

File: app/uk/gov/hmrc/traderservices/controllers/BaseJourneyController.scala [hunk 3]
```diff
@@ -16,64 +16,83 @@
...
+        .retrieve(credentials and authorisedEnrolments) {
+          case credentials ~ enrolments =>
+            val id = for {
+              enrolment  <- enrolments.getEnrolment(appConfig.authorisedServiceName)
+              identifier <- enrolment.getIdentifier(appConfig.authorisedIdentifierKey)
+            } yield identifier.value
+
+            Future.successful(credentials.map(_.providerId), id)
+          case _ => Future.successful(None, None)
+        }
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/uk/gov/hmrc/traderservices/controllers/BaseJourneyController.scala]: Return type for both were different, hence I used it this way. Now I have moved the authorised stuff to AuthActions class, to maintain consitency with authorised stuff.

File: it/uk/gov/hmrc/traderservices/services/MongoDBCachedJourneyServiceISpec.scala (hunk 1 above)
Comment (chandraAndem) [reply it/uk/gov/hmrc/traderservices/services/MongoDBCachedJourneyServiceISpec.scala]: Done.

**Review by ashwinirudrappa — APPROVED** [review]

**Pull request merged**